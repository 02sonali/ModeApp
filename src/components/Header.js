import React from 'react';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import {connect} from 'react-redux';
import * as actions from '../redux/actions/userActions';
import {bindActionCreators} from 'redux';

class Header extends React.Component {
  componentDidMount() {

    this.props.getLoginStatus();
  }

  logout() {
    this.props.logout();
  }

  render() {
    const isLoggedIn = this.props.isLoggedIn;
    return (
      isLoggedIn ? 
        <Navbar bg="dark" variant="dark">
          <Navbar.Brand href="/" className="mr-auto">Mode</Navbar.Brand>
          <Nav>
            <Nav.Link href="/articles">Articles</Nav.Link>
            <Nav.Link href="/new-article">New Article</Nav.Link>
            <Nav.Link href="#" onClick={() => this.logout()}>Logout</Nav.Link>
          </Nav>
        </Navbar> 
        :
        <Navbar bg="dark" variant="dark">
          <Navbar.Brand href="/" className="mr-auto">Mode</Navbar.Brand>
          <Nav>
            <Nav.Link href="/articles">Articles</Nav.Link>
            <Nav.Link href="/signin">Sign in</Nav.Link>
            <Nav.Link href="/signup">Sign up</Nav.Link>
          </Nav>
        </Navbar>
      )
  }
}

const mapStateToProps = function(state){
  // here we get the updated state value for user on dispatch action from register
  return {
    isLoggedIn: state.user.userId,
  }
}

const mapDispatchToProps = function (dispatch) {
  return bindActionCreators({
    getLoginStatus: actions.checkLoggedInUser,
    logout: actions.logout
  }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);