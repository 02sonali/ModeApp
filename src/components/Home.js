import React from 'react';
import Jumbotron from 'react-bootstrap/Jumbotron';
import Button from 'react-bootstrap/Button';
import {Link} from 'react-router-dom';
export const Home = () => {
   return <div>
      <Jumbotron>
         <h1>Mode</h1>
         <p>
            A place to express yourself and share your knowledge.<br/>
            Signup to start writing your own articles.
         </p>
         <p>
            <Link to="/articles">
               <Button variant="primary">Explore Articles</Button>
            </Link>
         </p>
      </Jumbotron>
   </div>
}

export default Home;