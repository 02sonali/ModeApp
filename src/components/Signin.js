import React from 'react';
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import {Link} from "react-router-dom";
import {useFormik} from 'formik';
import {useDispatch} from 'react-redux';
import { authStart } from '../redux/actions/userActions';

export const Signin = () => {
    const dispatch = useDispatch();
    const formik = useFormik({
        // validationSchema: {schema},
        initialValues: {
            email: '',
            password: ''
        },
        onSubmit: (values) => {
            console.log('form submission complete!!', values);
            dispatch(authStart(values));
        }
    })
    return <div className="container p-4">
        <h2>Sign in</h2>
        <Link to="/signup"> Need an account? </Link>
        <Form noValidate onSubmit={formik.handleSubmit} className="container p-4">
            <Form.Group controlId="email">
            <Form.Control type="email" placeholder="Email" onChange={formik.handleChange} onBlur={formik.handleBlur} value={formik.values.email}/>
                {/* <Form.Text className="text-muted">
                    We'll never share your email with anyone else.
                </Form.Text> */}
            </Form.Group>
            <Form.Group controlId="password">
            <Form.Control type="password" placeholder="Password" onChange={formik.handleChange} onBlur={formik.handleBlur} value={formik.values.password}/>
            </Form.Group>
            <Button variant="primary" type="submit">
                Sign in
            </Button>
        </Form>
   </div>
}

export default Signin;