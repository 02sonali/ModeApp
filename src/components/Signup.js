import React from 'react';
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import {Link} from "react-router-dom";
import {useFormik} from 'formik';
import {connect, useDispatch, useSelector} from 'react-redux';
import { registerUserStart } from '../redux/actions/userActions';

export const Signup = () => {
    const dispatch = useDispatch();
    const formik = useFormik({
        // validationSchema: {schema},
        initialValues: {
            username: '',
            email: '',
            password: ''
        },
        onSubmit: (values) => {
            console.log('form submission complete!!', values);
            dispatch(registerUserStart(values));
        }
    })
        
    return (
        <div className="container p-4">
        <h2>Sign up</h2>
        <Link to="/signin"> Have an account? </Link>
        <Form noValidate onSubmit={formik.handleSubmit} className="container p-4">
            <Form.Group controlId="username">
                <Form.Control type="text" placeholder="Username" onChange={formik.handleChange} onBlur={formik.handleBlur} value={formik.values.username}/>
                {/* <Form.Text className="text-muted">
                    We'll never share your email with anyone else.
                </Form.Text> */}
            </Form.Group>
            <Form.Group controlId="email">
                <Form.Control type="email" placeholder="Email" onChange={formik.handleChange} onBlur={formik.handleBlur} value={formik.values.email}/>
                {/* <Form.Text className="text-muted">
                    We'll never share your email with anyone else.
                </Form.Text> */}
            </Form.Group>
            <Form.Group controlId="password">
                <Form.Control type="password" placeholder="Password" onChange={formik.handleChange} onBlur={formik.handleBlur} value={formik.values.password}/>
            </Form.Group>
            <Button variant="primary" type="submit">
                Sign up
            </Button>
        </Form>
   </div>)
}

export default connect()(Signup);