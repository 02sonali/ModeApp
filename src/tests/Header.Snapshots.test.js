import React from 'react';
import renderer from 'react-test-renderer';
import {Header} from '../components/Header';

it("sets the header component", () => {
    const header = renderer.create(<Header></Header>);
    expect(header).toMatchSnapshot();
})