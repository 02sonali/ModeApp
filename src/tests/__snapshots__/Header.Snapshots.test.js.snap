// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets the header component 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <a
    className="mr-auto navbar-brand"
    href="/"
  >
    Mode
  </a>
  <div
    className="navbar-nav"
    onKeyDown={[Function]}
  >
    <a
      className="nav-link"
      data-rb-event-key="/articles"
      href="/articles"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Articles
    </a>
    <a
      className="nav-link"
      data-rb-event-key="/signin"
      href="/signin"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Sign in
    </a>
    <a
      className="nav-link"
      data-rb-event-key="/signup"
      href="/signup"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Sign up
    </a>
  </div>
</nav>
`;
